caikit_nlp/resources/pretrained_model/hf_auto_causal_lm.py

```diff
@@ -52,6 +52,7 @@ def build_task_tokenize_function(
         max_source_length: int,
         max_target_length: int,
         verbalizer: str,
+        task_ids: Union[None, int] = None,
```

Comment:
A couple of questions!

1. What is the motivation for doing this? I.e., is it to avoid the cost of setting task IDs for prompt tuning etc, or is it to not hardcode 0?
2. Should this be an `int`? I know we generally only use 0 for MPT with single target task, but If this is being made generic to not hardcore 0, an iterable is probably needed right?

Reply:
1. `task_ids` doesn't work for non peft models actually and we get error from `transformers` when we try to do training. So this would become problematic for fine-tuning. So I made this field optional to re-use the same tokenization logic.
2. I thought of iterable, but then we would have to figure out how to iterate of tasks, since presumably either we would want to map the tasks_ids with training input  or duplicate bunch of task_ids with each of the training input. For current use-cases, we actually need this to be hardcoded, so thought of moving this to prompt tuning layer itself 🤔

Reply:
Cool sounds great. Thanks!